solidity/test/LPRewardsTest.js

```diff
@@ -0,0 +1,219 @@
+
+      await fundKEEPReward(lpRewards.address, keepRewards)
+      await mintAndApproveWrappedTokens(
+        wrappedToken,
+        lpRewards.address,
+        staker1,
+        wrappedTokenStakerBallance
+      )
+
+      await lpRewards.setRewardDistribution(rewardDistribution)
+      await lpRewards.notifyRewardAmount(keepRewards, {from: rewardDistribution})
+
+      const wrappedTokensToStake = web3.utils
+        .toBN(2000)
+        .mul(tokenDecimalMultiplier)
+      await lpRewards.stake(wrappedTokensToStake, {from: staker1})
+
+      const future = (await time.latest()).add(time.duration.days(7))
+      await timeIncreaseTo(future)
+
+      const actualEarnings = (await lpRewards.earned(staker1)).div(
+        tokenDecimalMultiplier
+      )
+      const expectedEarnings = new BN(10000)
+      expect(actualEarnings).to.gte.BN(expectedEarnings.subn(precision))
+      expect(actualEarnings).to.lte.BN(expectedEarnings.addn(precision))
+
+      // 10,000 / 2,000 = 5 KEEP,
+      // Contract function will return 4999999999999999881 KEEP. Solidity
+      // does not have floating numbers and when dividing by 10^18, precision will
+      // not be on point.
+      const actualRewardPerToken = (await lpRewards.rewardPerToken()).div(
+        tokenDecimalMultiplier
+      )
+      const expectedRewardPerWrappedToken = new BN(5)
+      expect(actualRewardPerToken).to.gte.BN(
+        expectedRewardPerWrappedToken.subn(precision)
+      )
+      expect(actualRewardPerToken).to.lte.BN(
+        expectedRewardPerWrappedToken.addn(precision)
+      )
+    })
+
+    it("should be possible to withdraw rewards after staking wrapped tokens", async () => {
+      const rewardsAmount = new BN(5000)
+      const keepAllocated = rewardsAmount.mul(tokenDecimalMultiplier)
+      const wrappedTokenStakerBallance = web3.utils
+        .toBN(10000)
+        .mul(tokenDecimalMultiplier)
+
+      await fundKEEPReward(lpRewards.address, keepAllocated)
+      await mintAndApproveWrappedTokens(
+        wrappedToken,
+        lpRewards.address,
+        staker1,
+        wrappedTokenStakerBallance
+      )
+
+      await lpRewards.setRewardDistribution(rewardDistribution)
+      await lpRewards.notifyRewardAmount(keepAllocated, {from: rewardDistribution})
+
+      const wrappedTokensToStake = web3.utils
+        .toBN(2000)
+        .mul(tokenDecimalMultiplier)
+      await lpRewards.stake(wrappedTokensToStake, {from: staker1})
+
+      const future = (await time.latest()).add(time.duration.days(7))
+      await timeIncreaseTo(future)
+
+      // Withdraw wrapped tokens and KEEP rewards
+      await lpRewards.exit({from: staker1})
+
+      // Earned KEEP rewards for adding liquidity
+      const keepEarnedRewards = (await keepToken.balanceOf(staker1)).div(
+        tokenDecimalMultiplier
+      )
+      expect(keepEarnedRewards).to.gte.BN(rewardsAmount.subn(precision))
+      expect(keepEarnedRewards).to.lte.BN(rewardsAmount.addn(precision))
```

Comment:
I think we can hardcode `expectedEarnedRewards` so we can have just one exact equality check.

Reply:
How would that help? It's a time precision issue to my understanding.

Reply:
Precision error is very small, we had the same issue with `TestECDSARewards.js` and handled it the same way there. Besides, looks like `Unipool.js` also tests values from Solidy in a similar fashion by using `almost equal` approach.